### PR TITLE
Allow Antag names on spawn for ghost roles and antagonists

### PR DESCRIPTION
**Pull request by ReturnToZender**
## About The Pull Request
This pull request intends to make it so that players, when spawning as an antagonist or a ghost role, will use their alternate preference, allowing players to set an alt name for when they spawn as an antagonist.

Currently, this does not work, the preference does show up, but does not apply on either type of spawn. I'll need help to do the rest.

## Why It's Good For The Game
This prevents situations where John Medical from last round is the Lone Operative this round.
image
Progress so far.

## Changelog

:cl: ReturnToZender, Waterpig
add: Antag names
/:cl:

### CODE REVIEW

**Comment by ReturnToZender** [comment]
Progress update: Syndicate names now function properly
image

File: modular_skyrat/modules/contractor/code/datums/midround/antag_datum.dm [hunk 1]
```diff
@@ -23,6 +23,12 @@
 /datum/antagonist/contractor/on_gain()
 	forge_objectives()
 	. = ..()
+	//Bubber Edit: Adds automatic antag names
+	var/mob/living/carbon/human/person = owner.current
+	if(iscarbon(person))
+		person.apply_pref_name(/datum/preference/name/syndicate, person.client)
+		person.dna.update_dna_identity()
```
Comment (nevimer) [inline modular_skyrat/modules/contractor/code/datums/midround/antag_datum.dm:30]:
```suggestion
		if(person.client)
			person.apply_pref_name(/datum/preference/name/syndicate, person.client)
		person.dna.update_dna_identity()
```

**Review by nevimer — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/nukeop/nukeop.dm [hunk 2]
```diff
@@ -39,6 +39,11 @@
 		return
 
 	var/mob/living/carbon/human/operative = owner.current
+	//Bubber Edit: Adds automatic antag names
+	if(iscarbon(operative))
+		operative.apply_pref_name(/datum/preference/name/syndicate, operative.client)
+		operative.dna.update_dna_identity()
```
Comment (nevimer) [inline code/modules/antagonists/nukeop/nukeop.dm:45]:
```suggestion
	if(iscarbon(operative))
		if(operative.client)
			operative.apply_pref_name(/datum/preference/name/syndicate, operative.client)
		operative.dna.update_dna_identity()
```

**Comment by ReturnToZender** [comment]
PR closed due to getting stale, I'll get back to this later.

**Pull request closed without merging**